src/main/java/org/assertj/core/internal/Strings.java

```diff
@@ -518,6 +517,30 @@ public void assertContainsIgnoringCase(AssertionInfo info, CharSequence actual,
       throw failures.failure(info, shouldContainIgnoringCase(actual, sequence));
   }
 
+
+  /**
+   * Verifies that the given {@code CharSequence} contains the given strings, ignoring whitespaces.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the actual {@code CharSequence}.
+   * @param values the values to look for.
+   * @throws NullPointerException if the given sequence is {@code null}.
+   * @throws IllegalArgumentException if the given values is empty.
+   * @throws AssertionError if the given {@code CharSequence} is {@code null}.
+   * @throws AssertionError if the actual {@code CharSequence} does not contain the given sequence.
+   */
+  public void assertContainsIgnoringWhitespace(AssertionInfo info, CharSequence actual, CharSequence... values) {
+    doCommonCheckForCharSequence(info, actual, values);
+    String actualWithoutWhitespace = removeAllWhitespaces(actual);
+    Set<CharSequence> notFound = stream(values).filter(value -> !stringContains(actualWithoutWhitespace, removeAllWhitespaces(value)))
+      .collect(toCollection(LinkedHashSet::new));
```

Comment:
move `removeAllWhitespaces(value)` to a `map` operation before calling filter, something like:
```java
stream(values).map(this::removeAllWhitespaces)
              .filter(value -> !stringContains(actualWithoutWhitespace, value))
              .collect(toCollection(LinkedHashSet::new));
```